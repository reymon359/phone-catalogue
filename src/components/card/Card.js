import React from 'react';
import { Link } from 'react-router-dom';


const Card = (props) => {
    const {phone} = props;
    return (
        <Link to={`/details/${phone.id}`} phone={phone}>
            <div className="bg-white black dib br3 ma2 grow bw2 shadow-2-l pointer">
                <img
                    src={phone.imageFileName}
                    className="br3"
                    width="auto"
                    height="200"
                    alt="phones"
                />
                <div>
                    <h2>{phone.name}</h2>
                    <p className="b">{phone.price} 
                        <span role='img' aria-label='money-emoji'>ðŸ’²</span>
                    </p>
                </div>
            </div>
        </Link>
    );
};

export default Card;