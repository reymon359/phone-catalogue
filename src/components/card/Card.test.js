import { shallow } from 'enzyme';
import React from 'react';
import Card from './Card';

it('expects to render Card component', () => {
    const mockPhone = [
        {
            id: 1,
            name: 'Nokia',
            price: '2',
            imageFileName: 'http://localhost:3002/images/Honor_10.png'
        }
    ];
    expect(shallow(<Card phone={mockPhone} />)).toMatchSnapshot();
});
