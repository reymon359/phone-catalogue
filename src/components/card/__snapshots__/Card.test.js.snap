// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    phone={
      Array [
        Object {
          "id": 1,
          "imageFileName": "http://localhost:3002/images/Honor_10.png",
          "name": "Nokia",
          "price": "2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="bg-white black dib br3 ma2 grow bw2 shadow-2-l pointer"
      >
        <img
          alt="phones"
          className="br3"
          height="200"
          width="auto"
        />
        <div>
          <h2 />
          <p
            className="b"
          >
            <span
              aria-label="money-emoji"
              role="img"
            >
              ðŸ’²
            </span>
          </p>
        </div>
      </div>,
      "phone": Array [
        Object {
          "id": 1,
          "imageFileName": "http://localhost:3002/images/Honor_10.png",
          "name": "Nokia",
          "price": "2",
        },
      ],
      "to": "/details/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="phones"
            className="br3"
            height="200"
            width="auto"
          />,
          <div>
            <h2 />
            <p
              className="b"
            >
              <span
                aria-label="money-emoji"
                role="img"
              >
                ðŸ’²
              </span>
            </p>
          </div>,
        ],
        "className": "bg-white black dib br3 ma2 grow bw2 shadow-2-l pointer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "phones",
            "className": "br3",
            "height": "200",
            "src": undefined,
            "width": "auto",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <p
                className="b"
              >
                <span
                  aria-label="money-emoji"
                  role="img"
                >
                  ðŸ’²
                </span>
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <span
                    aria-label="money-emoji"
                    role="img"
                  >
                    ðŸ’²
                  </span>,
                ],
                "className": "b",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-label": "money-emoji",
                    "children": "ðŸ’²",
                    "role": "img",
                  },
                  "ref": null,
                  "rendered": "ðŸ’²",
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="bg-white black dib br3 ma2 grow bw2 shadow-2-l pointer"
        >
          <img
            alt="phones"
            className="br3"
            height="200"
            width="auto"
          />
          <div>
            <h2 />
            <p
              className="b"
            >
              <span
                aria-label="money-emoji"
                role="img"
              >
                ðŸ’²
              </span>
            </p>
          </div>
        </div>,
        "phone": Array [
          Object {
            "id": 1,
            "imageFileName": "http://localhost:3002/images/Honor_10.png",
            "name": "Nokia",
            "price": "2",
          },
        ],
        "to": "/details/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="phones"
              className="br3"
              height="200"
              width="auto"
            />,
            <div>
              <h2 />
              <p
                className="b"
              >
                <span
                  aria-label="money-emoji"
                  role="img"
                >
                  ðŸ’²
                </span>
              </p>
            </div>,
          ],
          "className": "bg-white black dib br3 ma2 grow bw2 shadow-2-l pointer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "phones",
              "className": "br3",
              "height": "200",
              "src": undefined,
              "width": "auto",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2 />,
                <p
                  className="b"
                >
                  <span
                    aria-label="money-emoji"
                    role="img"
                  >
                    ðŸ’²
                  </span>
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    <span
                      aria-label="money-emoji"
                      role="img"
                    >
                      ðŸ’²
                    </span>,
                  ],
                  "className": "b",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "money-emoji",
                      "children": "ðŸ’²",
                      "role": "img",
                    },
                    "ref": null,
                    "rendered": "ðŸ’²",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
